application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -122,18 +126,33 @@ private static boolean shouldBeArchived(MessageChannel channel, Instant archiveA
     }
 
     private void handleArchiveFlow(ThreadChannel threadChannel, MessageEmbed embed) {
-        Consumer<Throwable> handleFailure = error -> {
-            if (error instanceof ErrorResponseException) {
-                logger.warn("Unknown error occurred during help thread auto archive routine",
-                        error);
-            }
-        };
+
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithMention =
+                member -> threadChannel.sendMessage(member.get().getAsMention()).addEmbeds(embed);
+
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithoutMention =
+                member -> threadChannel.sendMessageEmbeds(embed);
 
         threadChannel.getGuild()
             .retrieveMemberById(threadChannel.getOwnerIdLong())
-            .flatMap(author -> threadChannel.sendMessage(author.getAsMention()).addEmbeds(embed))
-            .flatMap(any -> threadChannel.getManager().setArchived(true))
-            .queue(any -> {
-            }, handleFailure);
+            .mapToResult()
+            .flatMap(member -> {
+                if (member.isSuccess()) {
+                    return sendEmbedWithMention.apply(member);
+                }
+
+                logger.debug("Unable to mention user", member.getFailure());
+                return sendEmbedWithoutMention.apply(member);
+            })
+            .mapToResult()
+            .flatMap(sentEmbed -> {
+                if (sentEmbed.getFailure() instanceof ErrorResponseException) {
```

Comment:
Do we want to only do this on `ErrorResponseException`? What about any other type of error, they should be logged too.

Reply:
I suppose we can log it, goal was to handle errors inside JDA anything over network or external is logged by JDA no(correct me if i'm wrong)?

Reply:
Maybe but if we're being given a `getFailure()` we should use it.